### DISCUSSION TITLE
Critical CVE-2022-40674 vulnerability identified in kong 3.0.0 image

### DISCUSSION THREAD

**Discussion by ghost** [Kong/kong#9467] [Help] [answered]
The expat package in the kong 3.0.0 image is affected by a critical risk security issue CVE-2022-40674.

Please upgrade the expat package to remediate the security issue.

Installed Resource: expat 2.4.8-r0
Fixed Version: expat 2.4.9-r0

https://security.alpinelinux.org/vuln/CVE-2022-40674
https://nvd.nist.gov/vuln/detail/CVE-2022-40674

**Comment by shivpatel1** [comment]
Updating expat package and releasing might take some time. Meanwhile, can some one provide some information whether this CVE affects Kong in any way ? Or is it a false positive from Kong perspective ? If it affects Kong then is there any remediation steps we can take to make sure our environments are secure ?

**Comment by dndx** [comment, marked as answer]
Kong Gateway is not affected by this finding since its functionality is not dependent upon that library (no impact to CIA/no attack vector).

`expat` is used by the `git` and `perl` CLI shipped inside the image, and is included in the image to aid development. Kong does not invoke them with user provided input.

Kong Inc. uses the NIST Common Vulnerability Scoring System (CVSS) v3.0 to assign a risk ranking to each finding.

This has been classified as a LOW and has 180+ days for possible remediation.

You also have the path to create your images per our documentation:
* [Installation Options - v3.0.x | Kong Docs](https://docs.konghq.com/gateway/latest/install/)
* [Build your own Docker images - v3.0.x | Kong Docs](https://docs.konghq.com/gateway/latest/install/docker/build-custom-images/)
